### ISSUE TITLE
Timeframe for TensorRT export from HUB ??

### ISSUE THREAD

**Issue by barney2074** [ultralytics/hub#52]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

Hi,

The TensorRT export from HUB currently says' coming soon'
I was wondering what the timeframe for this is ?
Also, I'm assuming it will export a .engine file ?
![image](https://user-images.githubusercontent.com/25944617/177343275-36e3b4e3-9376-4707-b72f-9d65b35448e1.png)

thanks

Andrew

**Comment by glenn-jocher** [comment]
@barney2074 yes, the same applies to the YOLOv5 repo. TensorRT is by far the fastest GPU inference method (5x faster than PyTorch), so it's highly recommended to use this format for any GPU deployment. See our [GPU benchmarks](https://github.com/ultralytics/yolov5/pull/6963) for details.

### Colab Pro+ V100 High-RAM Results

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=0, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CUDA:0 (Tesla V100-SXM2-16GB, 16160MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 46.7/166.8 GB disk)

Benchmarks complete (458.07s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623                10.19
1             TorchScript        0.4623                 6.85
2                    ONNX        0.4623                14.63
3                OpenVINO           NaN                  NaN
4                TensorRT        0.4617                 1.89
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623                21.28
7     TensorFlow GraphDef        0.4623                21.22
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

Ideally the HUB TensorRT export would come with a GPU menu that you could select from, but right now only Enterprise GPUs are available in the cloud (i.e. Tesla, Volta, Ampere), so users for example would not be able to export for a 2080 or 3090 consumer GPU.

The most direct way for TensorRT export right now would just be to download the PyTorch model and then you could export locally with any GPU. Example TensorRT export commands are at the end of the YOLOv5 notebook here:

https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb
<img width="798" alt="Screenshot 2022-07-07 at 12 27 35" src="https://user-images.githubusercontent.com/26833433/177753076-9308fec2-233e-4ee1-b83b-2305cb9463ae.png">

**Comment by barney2074** [comment]
thanks Glenn

I did get the YOLOv5 model inferencing on my Jetson nano by following this:
[](https://wiki.seeedstudio.com/YOLOv5-Object-Detection-Jetson/)

but in hindsight probably easier to use YOLOv5 running in Docker on the nano

Andrew

**Comment by UltralyticsAssistant** [comment]
@barney2074 hello Andrew,

It sounds like you're getting a good handle on deploying YOLO models on the Jetson platform. Running YOLOv5 in a Docker container can indeed provide a more consistent environment, which might simplify deployment and development efforts.

As for the TensorRT export directly from the HUB, we appreciate your input on the ease and speed of deployment to devices such as the Nano, Xavier, or Orin. It’s certainly a priority to ensure simplicity for production deployment and we aim to make the process as straightforward as possible.

For updates and releases regarding the TensorRT export feature on the HUB, please keep an eye on the HUB repository and documentation, and feel free to reach out if you have any further questions or suggestions.

Your engagement with the project is valuable, and it helps us focus on features and improvements that matter to our users. Thanks for contributing to the conversation and helping us shape the future of Ultralytics HUB!

**Comment by CropWatchDevelopment** [comment]
Hello! Here from the future! Just wondering if there are any updates?
I know it may be a difficult task, I am just really excited to get an easy to export TensorRT model :D

**Comment by pderrenger** [comment]
Hello @CropWatchDevelopment,

Thank you for your enthusiasm and patience! We understand the excitement around the TensorRT export feature, and we're working diligently to bring this to the Ultralytics HUB.

As of now, the TensorRT export feature is still in development. Our team is focused on ensuring that the export process is as seamless and user-friendly as possible, while also addressing the complexities involved with different GPU environments.

In the meantime, you can still export your models to TensorRT locally using the YOLOv5 repository. Here's a quick example to get you started:

1. **Clone the YOLOv5 repository:**
    ```bash
    git clone https://github.com/ultralytics/yolov5
    cd yolov5
    ```

2. **Install the required dependencies:**
    ```bash
    pip install -r requirements.txt
    ```

3. **Export your model to TensorRT:**
    ```bash
    python export.py --weights yolov5s.pt --include engine
    ```

This will generate a `.engine` file that you can use for inference on your NVIDIA hardware.

For the latest updates, please keep an eye on our [Ultralytics HUB repository](https://github.com/ultralytics/hub) and [documentation](https://docs.ultralytics.com/hub/integrations/). We appreciate your support and feedback as we continue to improve our platform.

Feel free to reach out if you have any more questions or need further assistance. Your engagement helps us prioritize features that matter most to our community.
